src/proxy/outbound.rs

```diff
@@ -372,7 +372,8 @@ impl OutboundConnection {
         }
         // For case source client and upstream server are on the same node
         if !us.workload.node.is_empty()
-            && self.pi.cfg.local_node == Some(us.workload.node.clone())
+            && !self.pi.cfg.local_node.is_none()
```

Comment:
do we need this?

Reply:
No. Good catch. I was planning to unwrap local_node and wanted to ensure we didn't panic but it turns out that isn't necessary. Fix coming soon.